makefiles/suit.inc.mk

```diff
@@ -54,7 +58,7 @@ $(SUIT_MANIFEST): $(SUIT_MANIFEST_PAYLOADS) $(BINDIR_SUIT)
 	$(Q)rm -f $@.tmp
 
 $(SUIT_MANIFEST_SIGNED): $(SUIT_MANIFEST) $(SUIT_SEC)
-	$(Q)$(SUIT_TOOL) sign $(SUIT_TOOL_ARGS) -k $(SUIT_SEC) -m $(SUIT_MANIFEST) -o $@
+	$(Q)$(SUIT_TOOL) sign $(SUIT_TOOL_ARGS) -k $(word 1, $(SUIT_SEC)) -m $(SUIT_MANIFEST) -o $@
```

Comment:
Is this an assumption that the manifest is always to be signed with the first key in `SUIT_SEC`?

Reply:
So `SUIT_SEC` contains a list of keys to be embedded in the firmware to use for manifest verification.
But I assume our implementation only supports that a manifest is signed with one key.

If this line is based on the assumption that the key used for signing is always the first key, I would rather propose to introduce a variable `SUIT_SEC_SIGN` to specify the key to be used for signing.
By default it can be the first word in `SUIT_SEC`.

Reply:
yes that's the idea - good idea to make this more explicit / overwritable.
I also added some more comments.

Reply:
Even the password could be different for signing, in case you want to update the firmware with a new production key that you generate with `SUIT_SEC_PASSWORD`, but you want to sign the manifest with the old production key so that the node to be updated can verify the manifest successfully.

Reply:
The easiest solution would be to just switch to using OpenSSL to generate the key

```patch
--- a/makefiles/suit.base.inc.mk
+++ b/makefiles/suit.base.inc.mk
@@ -29,10 +29,14 @@ SUIT_PUB_HDR_DIR = $(dir $(SUIT_PUB_HDR))
 CFLAGS += -I$(SUIT_PUB_HDR_DIR)
 BUILDDEPS += $(SUIT_PUB_HDR)
 
+ifneq (,$(SUIT_SEC_PASSWORD))
+  OPENSSL_GENPKEY_ARGS += -aes-256-cbc
+endif
+
 $(SUIT_SEC): | $(CLEAN)
        $(Q)echo suit: generating key in $(SUIT_KEY_DIR)
        $(Q)mkdir -p $(SUIT_KEY_DIR)
-       $(Q)$(RIOTBASE)/dist/tools/suit/gen_key.py $@ $(SUIT_SEC_PASSWORD)
+       $(Q)openssl genpkey -algorithm ed25519 $(OPENSSL_GENPKEY_ARGS) -out $@
 
 %.pem.pub: %.pem
        $(Q)openssl ec -inform pem -in $< -outform pem -pubout -out $@
```

This will ask you to enter the key, which is the right thing to do anyway since you don't want to have the secret key in your shell history.
The only issue is that this leaves you with an empty key file on failure (passwords did not match) :confused:

Reply:
> which is the right thing to do anyway since you don't want to have the secret key in your shell history

this is solved by using openssl, yes.
But I thought about introducing a variable `SUIT_SEC_SIGN_PASSWORD` to be able to specify the old password for signing the manifest. I did not think of changing the key generation until you mentioned the thing with the shell history

Reply:
We can just test for the key file being empty and remove it if so.

Reply:
> But I thought about introducing a variable SUIT_SEC_SIGN_PASSWORD to be able to specify the old password

Now I get you, you redefined the meaning of `SUIT_SEC_PASSWORD` to be the signing key, because there is no variable for key creation anymore.

Reply:
Yea I think we should get rid of that 'password on the command line' thing.
While it sure is convenient, having it in the shell history kind of defeats the purpose of encrypting the key.

Reply:
For manifest signing it is still a variable, but instead of giving the password in the command line it could be exported in `.bashrc`. Reading it there from user input could be cumbersome